File: Factory/AbstractFactory/Java/src/PizzaStore/NewYorkPizzaStore.java
package PizzaStore;

import IngredientFactory.NewYorkPizzaIngredientFactory;
import IngredientFactory.PizzaIngredientFactory;
import Pizza.Pizza;
import Pizza.CheesePizza;
import Pizza.ClamPizza;
import Pizza.PepperoniPizza;
import Pizza.VeggiePizza;

public class NewYorkPizzaStore extends PizzaStore {
    @Override
    protected Pizza createPizza(String type) {
        Pizza pizza = null;
        PizzaIngredientFactory pizzaIngredientFactory = new NewYorkPizzaIngredientFactory();

        if(type.equals("cheese")){
            pizza = new CheesePizza(pizzaIngredientFactory);
            pizza.setName("New York Style Cheese Pizza");
        }

        if(type.equals("veggie")){
            pizza = new VeggiePizza(pizzaIngredientFactory);
            pizza.setName("New York Style Veggie Pizza");
        }

        if(type.equals("clam")){
            pizza = new ClamPizza(pizzaIngredientFactory);
            pizza.setName("New York Style Clam Pizza");
        }

        if(type.equals("pepperoni")){
            pizza = new PepperoniPizza(pizzaIngredientFactory);
            pizza.setName("New York Style Pepperoni Pizza");
        }

        return pizza;
    }
}

File: Factory/AbstractFactory/TypeScript/src/PizzaStore/NewYorkPizzaStore.ts
import Pizza from '../Pizza/Pizza';
import CheesePizza from '../Pizza/CheesePizza';
import PepperoniPizza from '../Pizza/PepperoniPizza';
import ClamPizza from '../Pizza/ClamPizza';
import VeggiePizza from '../Pizza/VeggiePizza';
import PizzaStore from './PizzaStore';
import PizzaIngredientFactory from '../IngredientFactory/PizzaIngredientFactory';
import NewYorkPizzaIngredientFactory from '../IngredientFactory/NewYorkPizzaIngredientFactory';

export default class NewYorkPizzaStore extends PizzaStore {
  protected createPizza(type: string): Pizza {
    let pizza: Pizza = null;
    const pizzaIngredientFactory: PizzaIngredientFactory =
      new NewYorkPizzaIngredientFactory();

    if (type === 'cheese') {
      pizza = new CheesePizza(pizzaIngredientFactory);
      pizza.setName('New York Style Cheese Pizza');
    }

    if (type === 'veggie') {
      pizza = new VeggiePizza(pizzaIngredientFactory);
      pizza.setName('New York Style Veggie Pizza');
    }

    if (type === 'pepperoni') {
      pizza = new PepperoniPizza(pizzaIngredientFactory);
      pizza.setName('New York Style Pepperoni Pizza');
    }

    if (type === 'clam') {
      pizza = new ClamPizza(pizzaIngredientFactory);
      pizza.setName('New York Style Clam Pizza');
    }

    return pizza;
  }
}
